src/status_im/contexts/communities/events.cljs

```diff
@@ -185,6 +185,37 @@
                      :on-success #(rf/dispatch [:communities/fetched-collapsed-categories-success %])
                      :on-error   #(log/error "failed to fetch collapsed community categories" %)}]}))
 
-(rf/reg-event-fx :communities/set-addresses-for-permissions
- (fn [{:keys [db]} [addresses]]
-   {:db (assoc-in db [:communities/addresses-for-permissions] addresses)}))
+(defn initialize-permission-addresses
+  [{:keys [db]}]
+  (let [accounts  (get-in db [:wallet :accounts])
+        addresses (set (map :address (vals accounts)))]
+    {:db (assoc db
+                :communities/previous-permission-addresses addresses
+                :communities/selected-permission-addresses addresses)}))
```

Comment:
`:communities/selected-permission-addresses` - This key will track the selected address for both passing it to the status-go API and displaying it in the UI.

`:communities/previous-permission-addresses` - This key is specifically for storing the previously selected state. It enables us to reset to the previous address when the user clicks 'Cancel' after making an edit. 

@ilmotta, I hope these new key names make more sense.

Reply:
It's perfect @ajayesivan. You even managed to come up with names with the same length, god mode.

Reply:
It was a pain 😓 I renamed it 4 or 5 times and finally ended up with this.

Reply:
<img src="https://github.com/status-im/status-mobile/assets/46027/a08ad88b-2534-420c-8012-27aa7d8936fa" width="350" />